### ISSUE TITLE
Machinectl takes over tty

### ISSUE THREAD

**Issue by tudurom** [Trundle/NixOS-WSL#29]
Hello, I'm trying to use GPG but it doesn't work because pinentry cannot open a tty.

```
$ echo GETPIN | pinentry -T /dev/pts/0
No $DBUS_SESSION_BUS_ADDRESS found, falling back to curses
OK Pleased to meet you
S ERROR gnome3.open_tty_for_read 83918849
ERR 83918849 Permission denied <Pinentry>
```

It works just fine if I start a new tty with `script /dev/null`:

```
$ script /dev/null
$ tty
/dev/pts/1
$ echo GETPIN | pinentry -T /dev/pts/1
```

Also there should be a way to have a systemd user session running, at least for dbus support. That way, graphical pinentry (GTK3) would work which is better.

**Comment by Reilithion** [comment]
I had a related issue when trying to enable `programs.ssh.startAgent` (in that having a user session might help?). Basically, the ssh agent wasn't starting. So I set the default user's password with `sudo passwd` and ran `sudo login`. Once I logged in, the ssh agent was running. Can you see if this workaround also has an effect on your issue?

**Comment by tudurom** [comment]
This worked very well! Thank you! The reason ssh agent works is because it seems that this also starts up systemd’s user session.

A curious thing I noticed is that even though before running `sudo login` the command `echo GETPIN | pinentry -T /dev/pts/0` would fail, causing gpg-agent to fail too, after running `sudo login` *and* closing the shell the command would also work, meaning that the tty was altered.

**Comment by tudurom** [comment]
There is one downside to this `sudo login` hack: graphical applications don't work anymore.

**Comment by tudurom** [comment]
This was fixed in 89c644e7178abc4abf51e4d1c82222ec18f9878a.
